src/AutoRest.CSharp/Common/Input/Inputs.cs

```diff
@@ -216,7 +216,8 @@ internal enum InputOperationParameterKind
         Client = 1,
         Constant = 2,
         Flattened = 3,
-        Grouped = 4,
+        Flatten = 4,
```

Comment:
what is the difference between Flattened = 3,  and Flatten = 4,

Reply:
Is it better to add some comments?  Those 2 names are quite confused.

Reply:
Why the newly-added optional is not at the end? Isn't this break anything?

Reply:
> Why the newly-added optional is not at the end? Isn't this break anything?

This is a internally used thing, we break ourselves therefore it is fine.
And in fact adding anything into an existing enum is a breaking change.

Reply:
I strongly suggest we change a name for this. Do not use the word `Flatten` here, it is really confusing.
I believe adding comment is only a workaround, if possible we should avoid any confusion on our code.
How about `Spreaded`?

Reply:
@ArcturusZhang How about `Spread`? `Spreaded` means the parameter is spreaded from other parameter. `Spread` can mean the parameter is to be spreaded.

Reply:
I think Spread is a good idea